examples/pod_log_node_proxy.rs

```diff
@@ -0,0 +1,80 @@
+use futures::TryStreamExt;
+use k8s_openapi::api::core::v1::Pod;
+use tracing::*;
+
+use futures::AsyncBufReadExt;
+use hyper::Uri;
+use kube::{
+    api::{Api, DeleteParams, ResourceExt},
+    core::{node_proxy::NodeProxyParams, subresource::LogParams},
+    Client, Config,
+};
+use serde_json::json;
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    tracing_subscriber::fmt::init();
+
+    let client = Client::try_default().await?;
+    let pods: Api<Pod> = Api::default_namespaced(client);
+
+    // create busybox pod that's alive for at most 30s
+    let p: Pod = serde_json::from_value(json!({
+        "apiVersion": "v1",
+        "kind": "Pod",
+        "metadata": {
+            "name": "example",
+            "labels": { "app": "kube-rs-test" },
+        },
+        "spec": {
+            "terminationGracePeriodSeconds": 1,
+            "restartPolicy": "Never",
+            "containers": [{
+              "name": "busybox",
+              "image": "busybox:1.34.1",
+              "command": ["sh", "-c", "for i in $(seq 1 5); do echo kube $i; sleep 0.1; done"],
+            }],
+        }
+    }))?;
+
+    match pods.create(&Default::default(), &p).await {
+        Ok(o) => assert_eq!(p.name_unchecked(), o.name_unchecked()),
+        Err(kube::Error::Api(ae)) => assert_eq!(ae.code, 409), // if we failed to clean-up
+        Err(e) => return Err(e.into()),                        // any other case if a failure
+    }
+
+    // wait for container to finish
+    tokio::time::sleep(std::time::Duration::from_secs(2)).await;
+
+    // Create a client for node proxy
+    let mut config = Config::infer().await?;
+    config.accept_invalid_certs = true;
+    config.cluster_url = "https://localhost:10250".to_string().parse::<Uri>().unwrap();
+    let client: Client = config.try_into()?;
```

Comment:
is this url hacking required to point to the kubelet port?

Reply:
Ah, so this is indeed necessary. You basically reroute yourself straight into the [kubelet metrics port](https://docs.k3s.io/installation/requirements#inbound-rules-for-k3s-nodes). I kind of mistakenly assumed it proxied through the apiserver (rather than using the apiserver as a cache).

This would be good to document on the `impl Client` part in `kube-client` with a strong message (i don't expect this type of access will be accessible for most users), and because of this, it is perhaps sensible to feature-flag it after all.

(If this feels like a lot, I am happy to help getting to the finish line, I do value the work you have done here, I am just trying to ensure we don't ship a feature that seems universally accessible without clear caveats).

Reply:
> Ah, so this is indeed necessary. You basically reroute yourself straight into the [kubelet metrics port](https://docs.k3s.io/installation/requirements#inbound-rules-for-k3s-nodes). I kind of mistakenly assumed it proxied through the apiserver (rather than using the apiserver as a cache).

In fact, it's not just the metrics port, but the whole kubelet port.

> This would be good to document on the impl Client part in kube-client with a strong message (i don't expect this type of access will be accessible for most users), and because of this, it is perhaps sensible to feature-flag it after all.
> (If this feels like a lot, I am happy to help getting to the finish line, I do value the work you have done here, I am just trying to ensure we don't ship a feature that seems universally accessible without clear caveats).

Not issue to put it behind a feature flag, I feel it's an advanced feature that requires high understanding, also it can lead to [privilege escalation](https://www.aquasec.com/blog/privilege-escalation-kubernetes-rbac/)
About the documentation, if you can provide me some text, happy to add it.

Reply:
Have added some example comments that should provide decent deterrents to all but the most advanced users. If you feel stronger caveats are useful, feel free to add them